Propse var reorganisation

gwarf opened the pull request:
Move vars from `vars/main.yml` to `defaults/main.yml` so that all the variables can easily be overridden when including the role in another playbook.
Related to #14

gwarf committed:
Merge vars/main.yml with defaults.yml

gwarf committed:
README: update variable configuration instructions

brucellino commented on README.md:
```diff
@@ -19,7 +19,7 @@ Requirements
 --------------
 
-Variables are used to tune the Rails application that runs Discourse. These are in a template, and set in the `defaults/main.yml` and `vars/main.yml` file. Please read them and set them accordingly in `vars/main.yml`
+Variables are used to tune the Rails application that runs Discourse. These are in a template, and set in the `defaults/main.yml` file. Please read them and set them accordingly in your playbook
```
Following our discussion in #14 I suggest we keep the original text, but remove `vars/main.yml` from change control.

brucellino commented on defaults/main.yml:
```diff
@@ -10,15 +10,27 @@ prerequisites:
-# Buffer size in MB
-db_shared_buffer_size: 256
-unicorn_workers: 2
+# discourse_hostname:
```
These are sane defaults, so I think they can stay.

brucellino commented on defaults/main.yml:
```diff
@@ -10,15 +10,27 @@ prerequisites:
-# change this - it makes no sense
-developer_emails: nobody@nowhere.com
-smtp_address: smtp.gmail.com
+developer_emails: 'brucellino@gmail.com'
```
This is not a sane default though

brucellino commented on defaults/main.yml:
```diff
@@ -10,15 +10,27 @@ prerequisites:
+smtp_address: smtp.mailgun.org
 smtp_port: 587
-smtp_username: admin@example.com
+smtp_username: postmaster@sandbox6c9929c31d284bdba0a154af210f7130.mailgun.org
```
neither is this.

brucellino commented on defaults/main.yml:
```diff
@@ -28,5 +40,26 @@ discourse_templates:
+# only the url is used in the template app.yml.j2.
+# The other fields are to help you remember what they are and why you
+# what them.
+discourse_plugins:
```
It doesn't make sense to have a long list of default plugins, every discourse is special :snowflake: but perhaps these are good examples of defaults. I'm ok with the list.

gwarf commented:
So as I was unsure I proposed to move all the contents from `vars/main.yml` to `defaults/main.yml`, but once the plan will be OK I will clean and use only the sane default values you mentioned. I was just confused by this embedding of non-easily overridable deployment-specific conf in the role.

brucellino committed:
Update README.md

Proposed description of where to put variables, with reference to Ansible docs.

brucellino reviewed (approved):
With a minor tweak of the README, I am staisfied

gwarf committed:
Use more generic defaults

gwarf committed:
Merge commit 'refs/pull/16/head' of github.com:AAROC/discourse-sso into propse_var_reorganisation

gwarf commented:
I've updated some of the defaults, let me know if it's OK with you and please squash/merge.
This should also close #14.

brucellino reviewed (approved):
The defaults make sense, so I'm ok with having the vars there.
